`invoke local` support for Ruby lambdas

dholdren opened the pull request:
<!--
1. Please check out and follow our Contributing Guidelines: https://github.com/serverless/serverless/blob/master/CONTRIBUTING.md
2. Do not remove any section of the template. If something is not applicable leave it empty but leave it in the PR
3. Please follow the template, otherwise we'll have to ask you to update it and it will take longer until your PR is merged
-->

## What did you implement:
Adds Ruby lambda support for `invoke local` 
Closes #5545

## How did you implement it:
I followed the pattern for calling Python lambda's via `invoke local`

## How can we verify it:

```
mkdir aws-ruby-example
cd aws-ruby-example
serverless create --template aws-ruby
serverless invoke local -f hello
```

Here's my result, using a local checkout of this branch:
```
➜  aws-ruby-example ~/work/serverless/bin/serverless invoke local -f hello
{"statusCode":200,"body":"\"Go Serverless v1.0! Your function executed successfully!\""}
```

## Todos:

- [x] Write tests
- [x] Write documentation
- [x] Fix linting errors
- [x] Make sure code coverage hasn't dropped
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO

rupakg commented on lib/plugins/aws/invokeLocal/invoke.rb:
```diff
@@ -0,0 +1,66 @@
+    exit 1
+  end
+
+  handler_path = "#{ARGV[0]}.rb"
```
`handler_path = ARGV[0]`

dholdren replied:
took care of this, since I switched to `require`

rupakg commented on lib/plugins/aws/invokeLocal/invoke.rb:
```diff
@@ -0,0 +1,66 @@
+  $LOAD_PATH << "."
+  $LOAD_PATH << "lib"
+  load(handler_path)
+  handler = method(handler_name)
```
Replace line 57-60 with:

```
require("./#{handler_path}")
```

dholdren replied:
changed this around a bit to support class methods

rupakg commented on lib/plugins/aws/invokeLocal/invoke.rb:
```diff
@@ -0,0 +1,66 @@
+  handler = method(handler_name)
+
+  context = FakeLambdaContext.new(**input.fetch('context', {}))
+  result = handler.call(event: input['event'], context: context)
```
`result = Object.new.send(handler_name, {event: input['event'], context: context})`

rupakg replied:
`load` will load and execute and code, and will also load the file each time, whereas `require` will only load and only do it once. Also, manipulating `$LOAD_PATH` adds another path to the Ruby load path that it needs to search hence increasing load times. 

The full code for `invoke.rb` after Line 50 looks like this:

```
handler_path = ARGV[0]
handler_name = ARGV[1]

input = JSON.load($stdin) || {}

require("./#{handler_path}")

context = LocalLambdaContext.new(**input.fetch('context', {}))
result = Object.new.send(handler_name, {event: input['event'], context: context})

puts result.to_json
```

Just my 2 cents.

dschep commented:
Thanks for the review @rupakg!

dholdren commented:
since AWS supports both a top-level method via `filebasename.foo` or a class method `filebasename.MyMod::MyClass.foo` (https://docs.aws.amazon.com/lambda/latest/dg/ruby-handler.html)  I just added support for both.

@rupakg I think I also addressed your other concerns

rupakg commented on lib/plugins/aws/invokeLocal/invoke.rb:
```diff
@@ -0,0 +1,67 @@
+  end
+
+  def memory_limit_in_mb
+    return '1024'
```
@dholdren minor nitpick - remove the unnecessary `return` from following 4 methods.

rupakg commented:
Awesome work @dholdren Thanks for getting invoke local Ruby integrated.
